pkg/model/field.go

```diff
@@ -58,7 +57,14 @@ func newField(
 	if shapeRef != nil {
 		shape = shapeRef.Shape
 	}
-	if shape != nil {
+
+	if cfg != nil && cfg.IsSecret {
+		gt = "*acktypes.SecretKeyReference"
+		gte = "acktypes.SecretKeyReference"
+		gtwp = "*acktypes.SecretKeyReference"
```

Comment:
This should be `ackv1alpha` not `acktypes` (see my comment on the runtime PR about putting SecretKeyReference into `apis/core/v1alpha1`)